src/lsst/cmservice/client/loaders.py

```diff
@@ -281,51 +349,54 @@ def campaign_cl(
         campaign : `Campaign`
             Newly created `Campaign`
         """
-        with open(campaign_yaml, encoding="utf-8") as fin:
-            config_data = yaml.safe_load(fin)
+        with open(campaign_yaml, encoding="utf-8") as f:
+            config_data = yaml.safe_load(f)
 
         try:
-            camp_config = config_data["Campaign"]
-            camp_config["parent_name"] = "DEFAULT"
+            manifest = config_data["Campaign"]
         except KeyError as msg:
             raise CMYamlParseError(
                 f"Could not find 'Campaign' tag in {campaign_yaml}",
             ) from msg
 
-        assert isinstance(camp_config, dict)
-
-        # flush out config_data with kwarg overrides
-        for key in ["name", "spec_name", "handler"]:
-            val = kwargs.get(key, None)
-            if val:
-                camp_config[key] = val
-
-        for key in ["data", "child_config", "collections", "spec_aliases"]:
-            camp_config.setdefault(key, {})
-            val = kwargs.get(key, None)
-            if val:
-                update_include_dict(camp_config[key], val)
-
-        spec_name = camp_config["spec_name"]
-
+        assert isinstance(manifest, dict)
+        if "data" not in manifest:
+            manifest["data"] = {}
+        # establish campaign namespace uuid
+        spec_name = manifest["spec_name"]
+        namespace = uuid5(DEFAULT_NAMESPACE, manifest["name"])
+        namespaced_spec_name = str(uuid5(namespace, spec_name))
+        manifest["spec_block_assoc_name"] = f"{namespaced_spec_name}#campaign"
+        # assert the the loaded campaign is using namespaced objects
+        manifest["data"]["namespace"] = str(namespace)
+        logger.info(f"""Loading campaign {manifest["name"]} with namespace {namespace}""")
+
+        # TODO deprecate this code path, are "already existing" specifications
+        #      allowed? Or is this allowed to invoke a "clone" operation where
```

Comment:
I can't think of any way this would work with `yaml_file=None`, since it would have to re-resolve all of the UUIDs to support a new name for the campaign. Effectively the yaml file is now the source of a clone, which I think is pretty reasonable in the near term.

Reply:
Agree, I don't think there's a need for heroic efforts to support piecemeal operations at the CLI loader level. If we prescribe a complete multi-doc YAML file as canonical for an entire campaign, then there is no server-side "clone" operation or even the selection of any spec blocks outside a specific campaign namespace except in some unit tests.